starlark/eval.go

```diff
@@ -1225,8 +1225,22 @@ func Call(thread *Thread, fn Value, args Tuple, kwargs []Tuple) (Value, error) {
 	fr.callable = c
 
 	thread.beginProfSpan()
+
+	// Use defer to ensure that panics from built-ins
+	// pass through the interpreter without leaving
+	// it in a bad state.
+	defer func() {
+		thread.endProfSpan()
```

Comment:
Nit: Readability risk: If someone adds more logic or cleanup following CallInternal, it'll be included in the profile span. We could 1) wrap the call to CallInternal in another function that just manages the profiling, or 2) add a comment below, or 3) ignore this if you think it's suitably obvious.

Reply:
That's true, but it's not worth complicating the control flow to avoid. And there's no law that says the prof span should include only the user code and not the other overheads that are inevitable when it is called.